### ISSUE TITLE
Tesla account not added

### ISSUE THREAD

**Issue by andriuskr** [OpenEVSE/ESP32_WiFi_V4.x#234]
Hello,

Trying to add tesla account (no MFA) to OpenEVSE. After saving credentials nothing happens, no info about Tesla SoC. {"count":0,"vehicles":[]}

**Comment by jeremypoulter** [comment]
Yeah, was looking at this over the weekend, Tesla have changed their login code again and the library we use has not been updated.

**Comment by andriuskr** [comment]
Is there any plans to update it ?

**Comment by jeremypoulter** [comment]
For sure, waiting on a fix in the upstream project: mseminatore/TeslaJS#261

**Comment by jeremypoulter** [comment]
Also will try to get #215 implemented so at least there is a work around.

**Comment by andriuskr** [comment]
Where on OpenEVSE web i can enter tokens ?

**Comment by glynhudson** [comment]
If you run the latest dev version, there is an advanced tab. 

![Screenshot 2021-11-07 02 03 15](https://user-images.githubusercontent.com/758844/140629538-e51fdf1f-f311-4fd3-885a-4c6775d47636.png)

Dev versions can now be downloaded from the releases page, marked as 'pre-release'. Be sure to select the correct release artefact to match your hardware.

**Comment by andriuskr** [comment]
Update to newest dev version. Vehicle tab is now gone. 
![552299EB-E5E7-41B0-9C1F-39AC12E3190D](https://user-images.githubusercontent.com/13429780/140643166-a138fabe-1229-4d35-b308-47d10064e85c.png)

**Comment by chris1howell** [comment]
Try clearing browser cache.

**Comment by andriuskr** [comment]
Tried even on new computer, not helping. Vehicle tab is gone

**Comment by jeremypoulter** [comment]
Think this is related to a resent merge, will fix shortly

**Comment by jeremypoulter** [comment]
Should be fixed now

**Comment by andriuskr** [comment]
Thanks, now it is working
![image](https://user-images.githubusercontent.com/13429780/140695020-a7636ba9-ca22-47ba-9cb9-e7a79808a082.png)

**Comment by andriuskr** [comment]
By the way, "Time to full charge" is not very accurate term. I have set to charge battery to 90%. It could fetch this from tesla acc and show not "Time to full charge", but "Time to 90% charge" or "Time to 80% charge" ?

**Comment by jeremypoulter** [comment]
Glad to hear that you have got it working. Will keep this open to fix the issue with the login. Can you raise the Time to full charge as a new issue?

**Comment by ve6rah** [comment]
Something is still causing me issues here. On both 4.1.2 and 4.1.4 I enter the tokens manually from the "Tesla Tokens" app (same app I use successfully for my home assistant integration) and all I ever see is "Fetching Vehicle Info" and it never gets any farther.
